### PR TITLE
v 2 with TS, demo, browserify and npm project

### PR DESCRIPTION
**Pull request by mariohmol**
Hi,

this project is very useful! 

I'm going to make an angular adapter for this .. and there is some fixes i need to do.. so was more comfortable to me to just fork it.

If you are interested to merge.. here we go =)

### CODE REVIEW

**Comment by eduardomourar** [comment]
@mariohmol , i liked this very much. Could you send an email to jsgantt.improved at gmail dot com, please? i would like to discuss this with you.

File: README.md [hunk 1]
```diff
@@ -1,21 +1,119 @@
-A fully featured gantt chart component built entirely in Javascript, CSS and AJAX. It is lightweight and there is no need of external libraries or additional images.
+[![Build Status](https://travis-ci.org/mariohmol/jsgantt-improved.svg?branch=master)](https://travis-ci.org/mariohmol/jsgantt-improved)
```
Comment (eduardomourar) [inline README.md]: just created `https://travis-ci.com/jsGanttImproved/jsgantt-improved` for this

File: package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/mariohmol/jsgantt-improved/issues"
+  },
+  "homepage": "https://github.com/mariohmol/jsgantt-improved#readme",
+  "dependencies": {},
+  "devDependencies": {
+    "@types/chai": "^4.1.5",
+    "@types/jasmine": "^3.3.0",
+    "chai": "^4.1.2",
```
Comment (eduardomourar) [inline package.json]: I had to install `browserify` and `nodemon` because didn't have them globally.

Comment (mariohmol) [reply package.json]: included it on readme

File: README.md (hunk 1 above)
Comment (mariohmol) [reply README.md]: changed

Comment (mariohmol) [reply README.md]: ok... i just saw about the docs folder.. did that

File: package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "start": "node server.js",
+    "dev": "nodemon server.js",
+    "build": "tsc",
+    "watch": "onchange 'src/**' '*.*' -- npm run build",
+    "watch:dist": "onchange 'src/**' '*.ts' -- npm run dist",
+    "watch:test": "onchange 'src/**/*.ts' '*.ts' -- npm run test",
+    "test": "node node_modules/.bin/protractor protractor.conf.js",
+    "test-unit": "mocha -r ts-node/register test/**.ts",
+    "browserify": "browserify dist/index.js --standalone JSGantt > jsgantt.js",
+    "dist": "npm run build && npm run browserify && cp src/jsgantt.css dist/ && cp jsgantt.js dist/ && cp jsgantt.js docs/ && cp src/jsgantt.css demo/ && echo 'DIST finished'",
```
Comment (eduardomourar) [inline package.json]: replace `demo/` with `docs/`

File: README.md [hunk 4]
```diff
@@ -1,21 +1,124 @@
-A fully featured gantt chart component built entirely in Javascript, CSS and AJAX. It is lightweight and there is no need of external libraries or additional images.
+[![Build Status](https://travis-ci.com/jsGanttImproved/jsgantt-improved.svg?branch=master)](https://travis-ci.com/jsGanttImproved/jsgantt-improved)
 
-You can view a live example at https://jsGanttImproved.github.io/jsgantt-improved
 
-Features include:
+A fully featured gantt chart component built entirely in Javascript, CSS and AJAX. It is lightweight and there is no need of external libraries or additional images. 
+
+
+![Demo Image](/demo/demo.png)
```
Comment (eduardomourar) [inline README.md]: replace `demo/` with `docs/`

File: README.md [hunk 5]
```diff
@@ -1,21 +1,124 @@
...
+
+
+Start using with including the files `jsgantt.js` and `jsgantt.css` that are inside `demo/` folder.
```
Comment (eduardomourar) [inline README.md]: replace `demo/` with `docs/`

File: README.md [hunk 6]
```diff
@@ -1,21 +1,124 @@
...
-  * Load Gantt from XML
+  * Load Gantt from XML and JSON
     * From external files (including experimental support for MS Project XML files)
     * From JavaScript Strings
-  * Support for Internationalization (all hard coded strings can be overridden)
+  * Support for Internationalization 
+
+## Documentation
+
+See the [Documentation](./Documentation.md) wiki page or the included ``demo/index.html`` file for instructions on use.
```
Comment (eduardomourar) [inline README.md]: replace `demo/` with `docs/`

File: package.json [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    "publish-npm": "npm run dist && npm publish",
+    "demo": "nodemon server.js",
+    "demo-full": "npm run dist && npm run demo",
+    "e2e-prepare": "npm i -g webdriver-manager && webdriver-manager update && ./node_modules/protractor/node_modules/webdriver-manager/bin/webdriver-manager update"
```
Comment (eduardomourar) [inline package.json]: based on this issue [here](https://github.com/angular/webdriver-manager/issues/216#issuecomment-293650564), we should just do `webdriver-manager update --gecko false` to see if improves build

Comment (mariohmol) [reply package.json]: what u mean about improve?

File: README.md (hunk 6 above)
Comment (mariohmol) [reply README.md]: did all the changes for demo and docs..

File: package.json (hunk 7 above)
Comment (eduardomourar) [reply package.json]: because of this error during build: https://travis-ci.com/jsGanttImproved/jsgantt-improved/builds/92611601

Comment (mariohmol) [reply package.json]: it because we changed to docs...

**Pull request merged**